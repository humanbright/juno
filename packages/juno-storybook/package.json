{
  "name": "@ringcentral/juno-storybook",
  "version": "1.0.0",
  "license": "MIT",
  "author": "RingCentral",
  "bugs": {
    "url": "https://github.com/ringcentral/juno/issues"
  },
  "homepage": "https://github.com/ringcentral/juno#readme",
  "repository": {
    "type": "git",
    "url": "https://github.com/ringcentral/juno.git"
  },
  "scripts": {
    "clean": "rm -rf node_modules",
    "clean-cache": "rm -rf node_modules/.cache"
  },
  "dependencies": {
    "@ringcentral/juno": "*",
    "moment": "^2.29.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "styled-components": "^5.3.3"
  }
}
