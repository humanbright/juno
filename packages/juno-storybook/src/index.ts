export * from './assets';
export * from './components';
export * from './decorators';
export * from './hooks';
export * from './isTestEnv';
export * from './utils';
