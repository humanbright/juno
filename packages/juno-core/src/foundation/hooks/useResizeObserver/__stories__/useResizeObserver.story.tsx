import React, { useRef } from 'react';

import { shadows, styled, useResizeObserver } from '@ringcentral/juno';
import { Title } from '@ringcentral/juno-storybook';
import { Meta, Story } from '@storybook/react';

export default {
  title: 'ðŸ”§ Foundation/Hooks/useResizeObserver',
  argTypes: {},
} as Meta;

const Wrapper = styled.div`
  width: 280px;
  resize: both;
  overflow: hidden;
  box-shadow: ${shadows('16')};
`;

export const useResizeObserverExample: Story<{}> = () => {
  const ref = useRef<HTMLDivElement>(null);

  useResizeObserver(ref, () => {
    console.log('resize');
  });

  return (
    <Wrapper ref={ref}>
      <Title>Resize here</Title>
    </Wrapper>
  );
};

useResizeObserverExample.args = {};
useResizeObserverExample.storyName = 'useResizeObserver';
