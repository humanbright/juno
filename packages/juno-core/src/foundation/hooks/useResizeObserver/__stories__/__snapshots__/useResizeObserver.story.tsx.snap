// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ðŸ”§ Foundation/Hooks/useResizeObserver useResizeObserver 1`] = `
.c1 {
  font-size: 1.5rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 32px;
  color: #121212;
}

.c0 {
  width: 280px;
  resize: both;
  overflow: hidden;
  box-shadow: 0px 8px 10px -5px rgba(0,0,0,0.2),0px 16px 24px 2px rgba(0,0,0,0.14),0px 6px 30px 5px rgba(0,0,0,0.12);
}

<div
  className="c0"
>
  <h5
    className="MuiTypography-root c1 MuiTypography-paragraph MuiTypography-displayBlock"
    data-color="neutral.f06"
    data-variant="headline1"
  >
    Resize here
  </h5>
</div>
`;
